test/dynamo/test_higher_order_ops.py

```diff
@@ -124,31 +128,40 @@ def _assert_wrap_fallback(self, func, args, setup=lambda: None):
     def _test_wrap_simple(
         self,
         func,
-        args,
+        args_generator,
         expected_num_wrap_args,
         expected_opcount=1,
         return_graph=False,
     ):
         # Given a `func` that has a single call to `wrap`,
         # we check that:
         # - there are no graph breaks
-        # - eager vs torch.compile has the same result
-        # - after dynamo capture, the wrap has the expected number of args
-        backend = EagerAndRecordGraphs()
-        cnt = CompileCounterWithBackend(backend)
-
-        expected = func(*args)
-        result = torch.compile(func, fullgraph=True, backend=cnt)(*args)
-
-        self.assertEqual(result, expected)
-
-        self.assertEqual(cnt.frame_count, 1)
-        self.assertEqual(cnt.op_count, expected_opcount)
-
-        self.assertEqual(len(backend.graphs), 1)
-        graph = backend.graphs[0]
-        wrap_node = find_first_node(graph, wrap)
-        self.assertEqual(len(wrap_node.args), expected_num_wrap_args)
+        # - eager vs torch.compile has the same result (correctness)
+        # - other compilation metrics, e.g, # of ops in the dynamo captured graph,
+        #   the wrap has the expected number of args, etc
+        #
+        # we have one or multiple runs through with each of the args from args_generator,
+        # and we will check:
+        # - correctness and no graph breaks for every run
+        # - other compilation metrics only for the first run, since automatic_dynamic_shapes
+        #   may compile another dynamic version graph for the later runs
+        graph = None
+        for i, args in enumerate(args_generator):
```

Comment:
Hmm, maybe `_test_wrap_simple` is doing too much. Do you have thoughts on alternate designs for the testing? The problem right now is that we wish to test a number of the wrap situations under different inputs to check that recompilation is correct. Maybe instead we want some OpInfo-like database that stores (HOO, sample inputs)